src/v/kafka/server/handlers/fetch.cc

```diff
@@ -702,6 +711,186 @@ class parallel_fetch_plan_executor final : public fetch_plan_executor::impl {
     }
 };
 
+class pid_controller {
+    // 0.1234 * scaling_factor = 1234
+    static constexpr int64_t scaling_factor = 10000;
+    static constexpr auto duration_unit = 1us;
+    static constexpr auto sample_rate = 250us;
+
+public:
+    pid_controller()
```

Comment:
can we unify that with `backlog_controller` ?

Reply:
It would be possible. However, it would require a newly written `pid_controller` class that supported both of the use cases and even then the new controller won't behave exactly the same as the existing ones. So we'd need testing of the new one for all uses cases as well. I don't think this is in scope for this project and will be not be unifying them unless someone is passionate about the idea.

Reply:
@mmaslankaprv looks like we aren't going to be able unify these as there are quite some fetch-specific behaviors encoded in the current controller, I'm going to resolve this but free-free to re-open if you have concerns.